# Pangenome-Graphs-in-Infectious-Disease
This repository contains simulated datasets and scripts used for the research paper titled "Pangenome Graphs in Infectious Disease: A Promising and Practical Implementation for Comprehensive Genetic Variation Analysis."
We have developed a pangenome graph pipeline for microbial genomics, consisting of graph construction using the PanGenome Graph Builder (PGGB) (https://github.com/pangenome/pggb), graph manipulation through the Optimized Dynamic Genome/Graph Implementation (ODGI)(https://github.com/pangenome/odgi), and calling for Next-Generation Sequencing (NGS) data using the VG toolkit(https://github.com/vgteam/vg)). 

The PGGB pipeline, a reference-free method, constructs pangenome graphs by employing all-to-all whole genome alignments with wfmash, graph induction via seqwish, and progressive normalization using smoothxg and gfaffix (Garrison Erik, 2023). ODGI facilitates graph manipulation tasks such as visualization, and extraction of distances among paths in the graph, enabling phylogenetic analysis (Guarracino et al., 2022). By utilizing the pangenome graph created with PGGB, it is possible to simultaneously identify various genetic variations, including structural variants, rearrangements, and small variants such as SNPs and insertions/deletions, through vg deconstruction. Additionally, We utilized the vg toolkit for analyzing NGS data against the graph for read mapping and variant calling (Garrison et al., 2018). 
